### ISSUE TITLE
[SCREENREADER]: Feedback button SHOULD be contained within a landmark

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6980]
#  [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The Foresee Feedback button **SHOULD** be contained within a landmark of the page structure, since all page content must be contained by landmarks. It is best practice to contain all content excepting skip links, within distinct regions such as the header, nav, main, and footer.

From [Deque University](https://dequeuniversity.com/rules/axe/3.5/region?application=AxeChrome): Screen reader users can navigate to a section based on its HTML element or ARIA Landmark. For example, you might use ARIA Landmarks to provide a simple replacement for a skip navigation link, though the replacement is only useful for users of screen readers. Sighted users or people using screen enlargers won't benefit from the addition, so it's not a good practice to substitute ARIA landmarks for skip navigation links altogether.

### Why it Matters
Navigating a web page is far simpler for screen reader users if the content splits between multiple high-level sections. Content outside of sections is difficult to find, and the content's purpose may be unclear.

Historically, HTML lacked some key semantic markers such as the ability to designate sections of the page as the header, navigation, main content, and footer. Using both HTML5 elements and ARIA landmarks in the same element is considered a best practice, but the future favors using native HTML5 element regions as browser support increases.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader or keyboard user, I want to navigate web content by landmarks and not miss anything. 

## Environment

* Operating System: any
* Browser: any
* Screenreading device: whatever is available to you
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/find-locations` in browser
2. Start screenreading device listed in Environment
3. Navigate the page using landmarks ([VoiceOver instructions](https://help.apple.com/voiceover/mac/10.15/#/vo35709) | [NVDA instructions - NVDA + F7](https://webaim.org/resources/shortcuts/nvda))
5. Verify that the Feedback button is not accessible via keyboard 

## Possible Fixes (optional)

Currently: Note in the **Code view** screenshot below that the Feedback button is contained in a div that is below the `<footer>` element.

Possible fixes: 
1. Move the code for the Feedback button into the `<footer>` element, just before the `</footer>`.
1. Alternatively, add `role="complementary"` to the `<div>` containing the Feedback button. Moving it into the footer would be preferable, but this works, too.

```diff
<!-- current code -->
! <div aria-label="Feedback, opens dialog" class="_acs _acsbadge--default _
!   acsbadge--medium _acsmiddleright _acsVertical_right _acsAnimate" role="button" 
!   style="display: block; overflow: hidden; background-color: rgb(0, 166, 210); 
!   visibility: visible;" tabindex="0">   
!     <svg aria-hidden="true" class="_acsBadgeImg _acsBadgeImg__aspark _acsBadgeImg__aspark--medium" viewBox="0 0 12 14" xmlns="http://www.w3.org/2000/svg"><title>Play Icon</title><g id="Layer_2"><g id="Layer_1-2"><path d="M0,0v14l12-7L0,0z" style="fill:#fff;"></path></g></g></svg>  
!     <span class="_acsBadgeLabel" style="color:#fff">Feedback</span> 
! </div>

<!-- possible fix -->
…
+ <div role="complementary" aria-label="Feedback, opens dialog" class="_acs _acsbadge--default _
+   acsbadge--medium _acsmiddleright _acsVertical_right _acsAnimate" role="button" 
+   style="display: block; overflow: hidden; background-color: rgb(0, 166, 210); 
+   visibility: visible;" tabindex="0">   
+     <svg aria-hidden="true" class="_acsBadgeImg _acsBadgeImg__aspark _acsBadgeImg__aspark--medium" viewBox="0 0 12 14" xmlns="http://www.w3.org/2000/svg"><title>Play Icon</title><g id="Layer_2"><g id="Layer_1-2"><path d="M0,0v14l12-7L0,0z" style="fill:#fff;"></path></g></g></svg>  
+     <span class="_acsBadgeLabel" style="color:#fff">Feedback</span> 
+  </div>
+ </footer>
```

## WCAG or Vendor Guidance (optional)

* [Deque University WCAG axe-core-3.2 All page content must be contained by landmarks](https://dequeuniversity.com/rules/axe/3.2/region)
* [W3C ARIA11: Using ARIA landmarks to identify regions of a page](https://www.w3.org/TR/WCAG20-TECHS/ARIA11.html)

### axe error view
![20200318-acs-feedback-landmark](https://user-images.githubusercontent.com/57469/76988857-cfd38200-691b-11ea-949f-9f4aab4c2b12.png)

### Code view
![20200318-acs-feedback-landmark-code](https://user-images.githubusercontent.com/57469/76990881-0ced4380-691f-11ea-8a12-d278988fcf95.png)

**Comment by laflannery** [comment]
No longer relevant, this is within the `main` landmark and accessible to screen reader and keyboard users.  Will close the issue in 1 week unless  otherwise told it needs to remain open.
